src/tools/wasm-reduce.cpp

```diff
@@ -851,7 +851,24 @@ struct Reducer
     }
     // the "opposite" of shrinking: copy a 'zero' element
     for (auto& segment : module->elementSegments) {
-      reduceByZeroing(segment.get(), first, 100, shrank);
+      reduceByZeroing(
+        segment.get(),
+        first,
+        [&](Expression* entry) {
+          if (entry->is<RefNull>()) {
+            // we don't need to replace a ref.null
+            return true;
+          } else if (!first || first->is<RefNull>()) {
```

Comment:
`first` cannot be null because of the check earlier.
```suggestion
          } else if (first->is<RefNull>()) {
```

Reply:
Done. Alpine didn't complain this time. Looks good now.